src/Microsoft.Azure.SignalR.Common/Endpoints/AccessKey.cs

```diff
@@ -14,20 +14,20 @@ internal class AccessKey
         public string Id => Key?.Item1;
         public string Value => Key?.Item2;
 
-        protected Tuple<string, string> Key { get; set; }
-
         public string Endpoint { get; }
-        public int? Port { get; }
+        public int Port { get; }
+
+        protected Tuple<string, string> Key { get; set; }
 
-        public AccessKey(string key, string endpoint, int? port) : this(endpoint, port)
+        public AccessKey(Uri endpoint, string key) : this(endpoint)
         {
             Key = new Tuple<string, string>(key.GetHashCode().ToString(), key);
         }
 
-        protected AccessKey(string endpoint, int? port)
+        protected AccessKey(Uri endpoint)
         {
-            Endpoint = endpoint;
-            Port = port;
+            Endpoint = $"{endpoint.Scheme}://{endpoint.Host}";
```

Comment:
what if endpoint contains subpath? If it is not supported, shall we do a pre-check and throw if it contains subpath? otherwise we silently drop some infomation